Widget/Fitness/FitnessWidget.swift

```diff
@@ -0,0 +1,238 @@
+        return "Open until " + dateFormatter.string(from: hours.1).replacingOccurrences(of: ":00", with: "") + ((showbusystring) ? (" • " + getBusyString(room: room)) : "")
+    } else {
+        return "Closed"
+    }
+}
+
+private func getHours (room: FitnessRoom) -> (Date, Date) {
+    let weekdayIndex: Int = (Calendar.current.component(.weekday, from: Date()) + 5) % 7
+    var hours: (Date, Date) {
+        let calendar = Calendar.current
+        let currentDate = Date()
+        let timeFormatter = DateFormatter()
+        timeFormatter.dateFormat = "HH:mm:ss"
+        let openTime = timeFormatter.date(from: room.open[weekdayIndex])!
+        let closeTime = timeFormatter.date(from: room.close[weekdayIndex])!
+
+        let openDate = calendar.date(bySettingHour: openTime.hour, minute: openTime.minutes, second: 0, of: currentDate)!
+        let closeDate = calendar.date(bySettingHour: closeTime.hour, minute: closeTime.minutes, second: 0, of: currentDate)!
+        return (openDate, closeDate)
+    }
+    return hours
+}
+
+private func isOpen (room: FitnessRoom) -> Bool {
+    let date = Date()
+    let hours = getHours(room: room)
+    return hours.0 <= date && date <= hours.1
+}
+
+
+
+private struct FitnessWidgetSmallView: View {
+    var rooms: [FitnessRoom]
+    var configuration: ConfigureFitnessWidgetIntent.Configuration
+    
+    
+    var body: some View {
+        VStack {
+            Text(rooms[0].name)
+                .font(.system(size: 13, weight: .semibold))
+                .padding(.top)
+            Text(getOpenString(room: rooms[0], showbusystring: false))
+                .font(.system(size: 11))
+                .foregroundStyle(isOpen(room: rooms[0]) ? .green : .blue)
+                .multilineTextAlignment(.center)
+                .fixedSize(horizontal: false, vertical: true)
+            MeterView(current: isOpen(room: rooms[0]) ? rooms[0].capacity : 0, maximum: 100.0, style: Color.blue, lineWidth: 6) {
+                VStack {
+                    Text("\(isOpen(room: rooms[0]) ? rooms[0].capacity : 0, specifier: "%.2f")%")
+                        .fontWeight(.bold)
+                    Text("capacity")
+                        .font(.system(size: 10, weight: .light, design: .default))
+                    
+                }
+            }
+            .frame(width: 90, height: 90)
+            
+            Text(lastUpdatedFormattedTime(date: rooms[0].last_updated))
+                .font(.system(size: 9))
+                .foregroundStyle(.gray)
+            
+            Spacer()
+        }.widgetPadding()
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+    }
+    
+}
+
+private struct FitnessWidgetMediumView: View {
+    var rooms: [FitnessRoom]
+    var configuration: ConfigureFitnessWidgetIntent.Configuration
+    
+    var body: some View {
+        VStack {
+            HStack {
+                Text(rooms[0].name)
+                    .font(.system(size: 12, weight: .semibold))
```

Comment:
Instead of making this a VStack with 2 HStacks inside, I think it should be the other way around. Right now, the top text is very badly centered with the graphs below (check out MPR, or even just first floor fitness).
I think the text on the left should be centered with the meter view (so those go in a vstack), and the text on the top right should go above the fitness graph (so those go in a vstack), and these two vstacks together are in a Hstack.